### PR TITLE
[MODULAR] Fixes glassblowing oversights

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request
This PR fixes two oversights present with glassblowing:

- The ability to work on glassworks while they are out of heat.
- The ability to work on multiple glassworks at once.

## How This Contributes To The Skyrat Roleplay Experience
This PR attempts to fix the issues presented by https://github.com/Skyrat-SS13/Skyrat-tg/pull/20895 of glassblowing products being able to easily be mass manufactured.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
No more blowing while no heat

image

image

No using multiple glass blowing tools at once:

image

image

</details>

## Changelog

:cl:
fix: glassblowing no longer works if the glass lacks heat
fix: glassblowing is unable to be done on multiple objects at once.
/:cl:

### CODE REVIEW

**Comment by vinylspiders** [comment]
Hi, I'm the one who did the recent refactor to glassblowing!

>     * The ability to work on glassworks while they are out of heat.

This was present before and may very well have been an oversight. I don't really know how glassblowing works IRL, so I just kept it so that the tools can be used on cooled glass. Don't really have any strong feelings towards this one.

>     * The ability to work on multiple glassworks at once.

As for this, I'm not sure this is an oversight per se. It was always the case that if you had multiple rods, you could multitask and juggle production between them.

I'm not sure removing that ability is really needed personally. Perhaps some people with more experience using it in rounds can chime in there.

**Comment by vinylspiders** [comment]
Don't see any issues with the code itself though for what it's worth. 👍

File: modular_skyrat/modules/primitive_production/code/glassblowing.dm [hunk 1]
```diff
@@ -366,12 +371,17 @@
 	if(!check_valid_table(user))
 		fail_message("You must be near a non-flammable table!", user)
 		return
-	if(!check_valid_tool(user, step_id))
-		return
+
+	var/datum/blower = check_valid_tool(user, step_id)
```
Comment (vinylspiders) [inline modular_skyrat/modules/primitive_production/code/glassblowing.dm]:
```suggestion
	var/atom/movable/tool_to_use = check_valid_tool(user, step_id)
```
Apart from maybe a more descriptive name for this.

File: modular_skyrat/modules/primitive_production/code/glassblowing.dm [hunk 2]
```diff
@@ -366,12 +371,17 @@
...
+	if(!blower)
+		in_use = FALSE
+		return FALSE
+
 	to_chat(user, span_notice("You begin to [step_id] [src]."))
 	if(!do_after(user, actioning_speed, target = src))
 		fail_message("You interrupt an action!", user)
-		return
+		REMOVE_TRAIT(blower, TRAIT_CURRENTLY_GLASSBLOWING, GLASSBLOWING_TRAIT)
```
Comment (vinylspiders) [inline modular_skyrat/modules/primitive_production/code/glassblowing.dm]:
```suggestion
		REMOVE_TRAIT(tool_to_use, TRAIT_CURRENTLY_GLASSBLOWING, GLASSBLOWING_TRAIT)
```

File: modular_skyrat/modules/primitive_production/code/glassblowing.dm [hunk 3]
```diff
@@ -366,12 +371,17 @@
 	if(!check_valid_table(user))
 		fail_message("You must be near a non-flammable table!", user)
 		return
-	if(!check_valid_tool(user, step_id))
-		return
+
+	var/datum/blower = check_valid_tool(user, step_id)
+	if(!blower)
```
Comment (vinylspiders) [inline modular_skyrat/modules/primitive_production/code/glassblowing.dm]:
```suggestion
	if(!tool_to_use)
```

File: modular_skyrat/modules/primitive_production/code/glassblowing.dm [hunk 4]
```diff
@@ -380,6 +390,7 @@
 			if(check_finished(glass))
 				glass.is_finished = TRUE
 
+	REMOVE_TRAIT(blower, TRAIT_CURRENTLY_GLASSBLOWING, GLASSBLOWING_TRAIT)
```
Comment (vinylspiders) [inline modular_skyrat/modules/primitive_production/code/glassblowing.dm]:
```suggestion
	REMOVE_TRAIT(tool_to_use, TRAIT_CURRENTLY_GLASSBLOWING, GLASSBLOWING_TRAIT)
```

**Comment by softcerv** [comment]
> > ```
> > * The ability to work on multiple glassworks at once.
> > ```
> 
> As for this, I'm not sure this is an oversight per se. It was always the case that if you had multiple rods, you could multitask and juggle production between them.
> 
> I'm not sure removing that ability is really needed personally. Perhaps some people with more experience using it in rounds can chime in there.

So, the for some context, this PR was made as an alternative/replacement to https://github.com/Skyrat-SS13/Skyrat-tg/pull/20895, part of the reason why that PR was made was because a certain glassblowing item (lenses I think) could be mass manufactured very easily, allowing for high amounts of credits to be farmed off of them. So having production limited to one product at a time could solve that problem while not completely tanking the value of glassblowing products. 

I haven't really talked with Jake, so it's hard for me to gauge if this was an intentional change or not, hence why I marked this as a fix.

File: modular_skyrat/modules/primitive_production/code/glassblowing.dm [hunk 5]
```diff
@@ -358,6 +358,11 @@
 /obj/item/glassblowing/blowing_rod/proc/do_glass_step(step_id, mob/user, actioning_speed, obj/item/glassblowing/molten_glass/glass)
 	if(!glass)
 		return
+
+	if(COOLDOWN_FINISHED(glass, remaining_heat))
+		to_chat(user, span_warning("The glass has cooled down and will require reheating to modify! "))
```
Comment (tf-4) [inline modular_skyrat/modules/primitive_production/code/glassblowing.dm]:
```suggestion
		balloon_alert(user, "glass too cool!")
```

File: modular_skyrat/modules/primitive_production/code/glassblowing.dm [hunk 6]
```diff
@@ -412,31 +423,40 @@
...
+ * If the correct tool is being used we return the tool. Otherwise we return `FALSE`
  */
 /obj/item/glassblowing/blowing_rod/proc/check_valid_tool(mob/living/carbon/human/user, step_id)
 	if(!istype(user))
 		return FALSE
 
-	var/obj/item/glassblowing/tool_path
+	if(step_id == STEP_BLOW || step_id == STEP_SPIN)
+		if(HAS_TRAIT(user, TRAIT_CURRENTLY_GLASSBLOWING))
+			balloon_alert(user, "you are already glassblowing!")
```
Comment (tf-4) [inline modular_skyrat/modules/primitive_production/code/glassblowing.dm]:
```suggestion
			balloon_alert(user, "already glassblowing!")
```

File: modular_skyrat/modules/primitive_production/code/glassblowing.dm [hunk 7]
```diff
@@ -412,31 +423,40 @@
...
-			tool_path = /obj/item/glassblowing/shears
+			used_tool = user.is_holding_item_of_type(/obj/item/glassblowing/shears)
 		if(STEP_JACKS)
-			tool_path = /obj/item/glassblowing/jacks
+			used_tool = user.is_holding_item_of_type(/obj/item/glassblowing/jacks)
 
-	if(user.is_holding_item_of_type(tool_path))
-		return TRUE
+	if(!used_tool)
+		balloon_alert(user, "you don't have the right tool!")
```
Comment (tf-4) [inline modular_skyrat/modules/primitive_production/code/glassblowing.dm]:
```suggestion
		balloon_alert(user, "need the right tool!")
```

File: modular_skyrat/modules/primitive_production/code/glassblowing.dm [hunk 8]
```diff
@@ -412,31 +423,40 @@
...
+		return FALSE
 
-	balloon_alert(user, "need to be holding [initial(tool_path.name)]!")
-	in_use = FALSE
-	return FALSE
+	if(HAS_TRAIT(used_tool, TRAIT_CURRENTLY_GLASSBLOWING))
+		balloon_alert(user, "[used_tool] is currently in use.")
```
Comment (tf-4) [inline modular_skyrat/modules/primitive_production/code/glassblowing.dm]:
```suggestion
		balloon_alert(user, "already in use!")
```

**Review by tf-4 — CHANGES_REQUESTED** [review]
just a couple stylistic changes with the balloon alerts

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**